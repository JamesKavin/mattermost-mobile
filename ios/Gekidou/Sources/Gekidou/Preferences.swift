//
//  Preferences.swift
//  Gekidou
//
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
//
import Foundation

let appGroupId = Bundle.main.infoDictionary!["AppGroupIdentifier"] as? String

public class Preferences: NSObject {
    private let userDefauls = UserDefaults(suiteName: appGroupId)
    
    @objc public static let `default` = Preferences()
    
    override private init() {}
    
    public func object(forKey name: String) -> Any? {
        return userDefauls?.object(forKey: name)
    }
    
    public func removeObject(forKey name: String) {
        userDefauls?.removeObject(forKey: name)
    }
    
    public func set(_ value: Any?, forKey key: String) {
        userDefauls?.set(value, forKey: key)
    }
}
