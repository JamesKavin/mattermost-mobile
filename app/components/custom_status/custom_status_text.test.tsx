// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import CustomStatusText from '@components/custom_status/custom_status_text';
import Preferences from '@constants/preferences';
import {render} from '@test/intl-test-helper';

describe('components/custom_status/custom_status_text', () => {
    const baseProps = {
        text: 'In a meeting',
        theme: Preferences.THEMES.denim,
    };

    it('should match snapshot', () => {
        const wrapper = render(
            <CustomStatusText
                {...baseProps}
            />,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with empty text', () => {
        const wrapper = render(
            <CustomStatusText
                {...baseProps}
                text={''}
            />,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
