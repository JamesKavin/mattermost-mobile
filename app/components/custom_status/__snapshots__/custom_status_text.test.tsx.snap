// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/custom_status/custom_status_text should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "rgba(63,67,80,0.5)",
        "fontSize": 17,
        "includeFontPadding": false,
        "textAlignVertical": "center",
      },
      undefined,
    ]
  }
>
  In a meeting
</Text>
`;

exports[`components/custom_status/custom_status_text should match snapshot with empty text 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "rgba(63,67,80,0.5)",
        "fontSize": 17,
        "includeFontPadding": false,
        "textAlignVertical": "center",
      },
      undefined,
    ]
  }
>
  
</Text>
`;
